### PR TITLE
Fixes #5366 - Vectorcompat all the things

### LINKED ISSUE #5366: VectorDrawableCompat all the things
We've been having issues with vector drawables on Android SDK 21-24 due to issues that have since been fixed in newer Android versions. We should start using the compat library everywhere we use vector icons so it handles those older devices for us.

[Using vector assets in Android apps](https://medium.com/androiddevelopers/using-vector-assets-in-android-apps-4318fd662eb9)

Related:
- #5359
- #5316
- #4702

### PR DESCRIPTION
**Pull request by icenine457**
Use various *compat class methods, *compat attributes in declarative layouts, for SDK 21-24 support.

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/collections/CollectionCreationUIView.kt [hunk 1]
```diff
@@ -137,7 +137,7 @@ class CollectionCreationUIView(
                 view.bottom_button_bar_layout.setOnClickListener(null)
                 view.bottom_button_bar_layout.isClickable = false
 
-                val drawable = view.context.getDrawable(R.drawable.ic_close)
+                val drawable = ContextCompat.getDrawable(view.context, R.drawable.ic_close)
```
Comment (NotWoods) [inline app/src/main/java/org/mozilla/fenix/collections/CollectionCreationUIView.kt]: `ContextCompat` isn't sufficient for vector compat. Internally it has the following if statement:
```java
if (Build.VERSION.SDK_INT >= 21) {
    return context.getDrawable(id);
} else ...
```

You must use [`AppCompatResources.getDrawable`](https://medium.com/androiddevelopers/using-vector-assets-in-android-apps-4318fd662eb9) instead.

File: app/src/main/java/org/mozilla/fenix/home/SearchView.kt [hunk 2]
```diff
@@ -8,15 +8,16 @@ import android.content.Context
 import android.graphics.drawable.TransitionDrawable
 import android.util.AttributeSet
 import android.widget.FrameLayout
+import androidx.core.content.res.ResourcesCompat
 import org.mozilla.fenix.R
 
 class SearchView(context: Context, attrs: AttributeSet) : FrameLayout(context, attrs) {
     private val lightDrawable =
-        resources.getDrawable(R.drawable.home_search_background_normal, context.theme)
+        ResourcesCompat.getDrawable(resources, R.drawable.home_search_background_normal, context.theme)
```
Comment (NotWoods) [inline app/src/main/java/org/mozilla/fenix/home/SearchView.kt]: Like `ContextCompat`, `ResourcesCompat` doesn't work with vector compat. (Why are there so many ways to get a drawable? 🤷‍♀) Use `AppCompatResources.getDrawable` with the context property on the view.

File: app/src/main/res/layout/collection_home_list_row.xml [hunk 3]
```diff
@@ -24,7 +24,7 @@
         android:layout_marginStart="16dp"
         android:layout_marginTop="16dp"
         android:importantForAccessibility="no"
-        android:src="@drawable/ic_tab_collection"
+        app:srcCompat="@drawable/ic_tab_collection"
```
Comment (NotWoods) [inline app/src/main/res/layout/collection_home_list_row.xml]: 🎉

**Comment by icenine457** [comment]
Thanks for the feedback! I'll incorporate these changes and re-submit.

**Comment by boek** [comment]
Hey @icenine457, I'm going to close this for now. Feel free to open a new PR when its ready :)

**Pull request closed without merging**